### PR TITLE
sx: Calculate insertIndex inside loop in injectRuntimeStyles

### PR DESCRIPTION
**Pull request by tbergq**
This was defined outside the loop before, it was ok but confusing.
It is easy to think that this works like an array and that you are
overwriting index, though it is just adding and pushing the other down.
Also added a test for that the runtime styles are added correctly

### CODE REVIEW

File: src/sx/src/__tests__/injectRuntimeStyles.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  sx.create({
+    test: {
+      opacity: 0,
+      height: '100px',
+    },
+  });
+
+  expect(styleTag?.sheet?.cssRules).toHaveLength(2);
+  expect(styleTag.sheet.cssRules).toMatchInlineSnapshot(`
```
Comment (mrtnzlml) [inline src/sx/src/__tests__/injectRuntimeStyles.test.js:21]: Why are the optional chains required on line 20 and not on line 21? 🤔

**Review by mrtnzlml — APPROVED** [review]
Thanks!

File: src/sx/src/__tests__/injectRuntimeStyles.test.js (hunk 1 above)
Comment (tbergq) [reply src/sx/src/__tests__/injectRuntimeStyles.test.js:21]: Yeah, good question, I didn't notice. 

it seems like it was not necessary after adding `globalThis.document`, so I guess that makes the styleTag any?

I will clean it up.

Comment (mrtnzlml) [reply src/sx/src/__tests__/injectRuntimeStyles.test.js]: Might be. `globalThis` and `global` are type of `any`. Sometimes it's ok to access these global objects directly and just explain it to Eslint like so: `/* global document */`

Comment (tbergq) [reply src/sx/src/__tests__/injectRuntimeStyles.test.js]: Yeah, that also works 🤔 

My idea was showing lint-error even if I added global document, but I see it passes in the terminal.

**Pull request merged**